assets/js/googlesitekit/widgets/components/WidgetAreaRenderer.js

```diff
@@ -102,69 +97,74 @@ const resizeClasses = ( classNames, counter ) => {
+		counter += WIDTH_GRID_COUNTER_MAP[ width ];
 
-			const width = widget.width;
+		// If counter is exactly 12, the next widget is going to be in a new row.
+		if ( counter % 12 === 0 ) {
+			counter = 0;
+		}
 
-			// Increase column counter based on width.
-			counter += WIDTH_GRID_COUNTER_MAP[ width ];
+		// If counter is going above 12, this widget is too wide for the current row.
+		// So it's going to be the first widget in the next row instead.
+		if ( counter > 12 ) {
+			counter -= WIDTH_GRID_COUNTER_MAP[ width ];
 
-			// If counter is exactly 12, the next widget is going to be in a new row.
-			if ( counter % 12 === 0 ) {
-				counter = 0;
+			// If the column count without the overflowing widget is exactly 9, expand
+			// the widths of these widgets slightly to fill the entire 12 columns.
+			if ( counter === 9 ) {
+				[ classNames, counter ] = resizeClasses( classNames, counter );
 			}
 
-			// If counter is going above 12, this widget is too wide for the current row.
-			// So it's going to be the first widget in the next row instead.
-			if ( counter > 12 ) {
-				counter -= WIDTH_GRID_COUNTER_MAP[ width ];
+			// See above, initial counter for the next row of widgets.
+			counter = WIDTH_GRID_COUNTER_MAP[ width ];
+		}
 
-				// If the column count without the overflowing widget is exactly 9, expand
-				// the widths of these widgets slightly to fill the entire 12 columns.
-				if ( counter === 9 ) {
-					[ classNames, counter ] = resizeClasses( classNames, counter );
-				}
+		// Actually set the class for the current widget. This must be set after
+		// potentially resizing classes, since in that case this will be the overflowing
+		// widget which should NOT be adjusted because it will be in the next row.
+		classNames[ i ] = WIDTH_GRID_CLASS_MAP[ width ];
+	} );
 
-				// See above, initial counter for the next row of widgets.
-				counter = WIDTH_GRID_COUNTER_MAP[ width ];
-			}
+	if ( counter === 9 ) {
+		[ classNames, counter ] = resizeClasses( classNames, counter );
+	}
 
-			// Actually set the class for the current widget. This must be set after
-			// potentially resizing classes, since in that case this will be the overflowing
-			// widget which should NOT be adjusted because it will be in the next row.
-			classNames[ i ] = WIDTH_GRID_CLASS_MAP[ width ];
-		} );
+	return classNames;
+};
 
-		if ( counter === 9 ) {
-			[ classNames, counter ] = resizeClasses( classNames, counter );
-		}
+const WidgetAreaRenderer = ( { slug } ) => {
+	const widgetArea = useSelect( ( select ) => select( STORE_NAME ).getWidgetArea( slug ) );
+	const widgets = useSelect( ( select ) => select( STORE_NAME ).getWidgets( slug ) );
+
+	// Verify that widgets are active (do not render null)
+	const isActiveWidget = ( widget ) =>
+		widgets.some( ( item ) => item.slug === widget.slug ) &&
+		typeof widget.component === 'function' &&
+		widget.component.prototype.render
+			? new widget.component( {} ).render()
+			: widget.component( {} );
```

Comment:
I think we can move this function out of the component, it doesn't need to be declared inside the component itself. Furthermore, for readability I'd suggest not using shorthand syntax for this one because it spans multiple lines and conditionals, so I think it makes it unnecessarily hard to digest.

Reply:
Abstracted the component out and put individual conditionals into variables to increase readability.